### PR TITLE
Make elasticsearch discovery supports IPv6

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Make elasticsearch discovery supports IPv6.

As IPv6 address contains `:`(one of the [special value](https://www.elastic.co/guide/en/elasticsearch/reference/6.8/modules-network.html#network-interface-values)), it should be bracketed.

Without this PR, it won't resolve the IPv6 address correctly and will fail to start.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
addons: elasticsearch discovery supports IPv6
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go [hunk 1]
```diff
@@ -52,7 +53,13 @@ func flattenSubsets(subsets []corev1.EndpointSubset) []string {
 	ips := []string{}
 	for _, ss := range subsets {
 		for _, addr := range ss.Addresses {
-			ips = append(ips, fmt.Sprintf(`"%s"`, addr.IP))
+			if ip := net.ParseIP(addr.IP); ip != nil {
```
Comment (aojea) [inline cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go]: nit, you have a function in k/utils that is doing this
https://github.com/kubernetes/utils/blob/6ca3b61696b65b0e81f1a39b4937fc2d2994ed6a/net/net.go#L121

Comment (SataQiu) [reply cluster/addons/fluentd-elasticsearch/es-image/elasticsearch_logging_discovery.go]: Good! Thanks @aojea

**Review by coffeepac — APPROVED** [review]
/approve
/lgtm

Thanks!

**Pull request merged**